lib/neo4j/core/cypher_session/responses/bolt.rb

```diff
@@ -52,10 +52,16 @@ def wrap_entity(entity_data)
 
           def extract_message_groups(flush_messages_proc)
             fields_messages = flush_messages_proc.call
+
             validate_message_type!(fields_messages[0], :success)
 
             result_messages = []
-            while (messages = flush_messages_proc.call)[0].type != :success
+            messages = nil
+
+            loop do
+              messages = flush_messages_proc.call
+              next if messages.nil?
```

Comment:
@cheerfulstoic This is pretty critical. This line avoids the bug, but I need confirmation.

Can you take a look?